### ISSUE TITLE
Compile Error using Saturation mode Optimizer

### ISSUE THREAD

**Issue by selwyn96** [fastmachinelearning/hls4ml#545]
Hi everyone, I am working on a branch to add Alveo support for hls4ml. I have my code up to date with the master branch and am currently running tests to make sure everything works fine. 
I am seeing a error when including the code `hls4ml.model.optimizer.get_optimizer('output_rounding_saturation_mode').configure(layers=['Activation'], rounding_mode='AP_RND', saturation_mode='AP_SAT')`
when configuring my hls4ml model. This error appears during `hls_model.compile()`
The error error can be removed by adding `hls4ml.model.optimizer.get_optimizer('output_rounding_saturation_mode').configure(layers=[])` before compile without compromising the accuracy. 
Why is this happening?
![error](https://user-images.githubusercontent.com/31563706/167224775-7fab6dcf-21f7-41d4-a3b8-b5a2e3988b24.png)

**Comment by BertSam** [comment]
I have no idea, but I'm trying to compile for 1 day now and your workaround probably save me a lot of time ! 

Still a weird bug that would required to be fixed.
